### DISCUSSION TITLE
Copilot Feedback

### DISCUSSION THREAD

**Discussion by CryptxzzX** [community/community#18801] [Copilot Conversations]
Just a small thing I've noticed after using copilot for some months now.
I noticed that copilot's suggestions don't seem to be as big/complex as they use to be.
I also feel like it's gotten worse at using previous code as context, I'm not sure if this is just me experiencing this or if it's part of an update, with this I've also noticed it tends to give me a lot more comments rather then actual code.

**Comment by thomscoder** [comment]
Ive been an avid user of Copilot for some months now and overall I'm experiencing same performances. Yes there still is some things that can be improved. 
I code most of the time in JS and Typescript, but I've seen great suggestions even in my small C++ projects.
In which languages are you experiencing your problems?

**Comment by davecheney** [comment]
Please try adding this fragment to Copilot's `settings.json`
```
    "github.copilot.advanced": {
        "debug.overrideEngine": "copilot-logit",
    },
```
And please let me know if that helped.

**Comment by dannypike** [comment]
I am finding the same. It was great for a while (C++ and python) but now the answers are "idiotic" at best, parroting the question that I ask or just plain incorrect.

What did you do to it?

For example, to the question __*write a simple "Hello World" program in Python3*__, I get:
```
Suggestion 1
write a simple "Hello World" program in Python3
print

Accept suggestion 1
Suggestion 2
write a simple "Hello World" program in Python 3
print("Hello World")
 
 This is the simplest Python program. 
 Python program to add two numbers 
 #!/usr/bin/env python3
# write a simple program to add two numbers
num1 = 20
num2 = 30
sum = num1 + num2
print("The sum of {0} and {1} is {2}".format(num1, num2, sum))
 
 Python program to find the square root of a number 
 #!/usr/bin/env python3
# write a simple program to find the square root of a number
num = 16
sqrt = num ** 0.5
print("The square root of {0} is {1}".format(num, sqrt))
 
 Python program to find the factorial of a number 
 #!/usr/bin/env python3
# write a simple program to find the factorial of a number
num = 5
factorial = 1
for i in range(1, num + 1):
    factorial *= i
print("The factorial of {0} is {1}".format(num, factorial))
 
 Python program to find the largest number among three numbers 
 #!/usr/bin/env python3
# write a simple program to find the largest number among three numbers
num1 = 10
num2 = 20
num3 = 30

Accept suggestion 2
```
